index/getAutoRefresh_test.go

```diff
@@ -0,0 +1,71 @@
+package index_test
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+
+	"github.com/kuzzleio/sdk-go/connection/websocket"
+	"github.com/kuzzleio/sdk-go/index"
+	"github.com/kuzzleio/sdk-go/internal"
+	"github.com/kuzzleio/sdk-go/kuzzle"
+	"github.com/kuzzleio/sdk-go/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetAutoRefreshNull(t *testing.T) {
+	k, _ := kuzzle.NewKuzzle(&internal.MockedConnection{}, nil)
+	i := index.NewIndex(k)
+	_, err := i.GetAutoRefresh("")
+	assert.NotNil(t, err)
+}
+
+func TestGetAutoRefreshQueryError(t *testing.T) {
+	c := &internal.MockedConnection{
+		MockSend: func(query []byte, options types.QueryOptions) *types.KuzzleResponse {
+			return &types.KuzzleResponse{Error: &types.KuzzleError{Message: "error"}}
+		},
+	}
+	k, _ := kuzzle.NewKuzzle(c, nil)
+	i := index.NewIndex(k)
+	_, err := i.GetAutoRefresh("index")
+	assert.NotNil(t, err)
+}
+
+func TestGetAutoRefresh(t *testing.T) {
+	type ackResult struct {
+		Acknowledged       bool
+		ShardsAcknowledged bool
+	}
+
+	c := &internal.MockedConnection{
+		MockSend: func(query []byte, options types.QueryOptions) *types.KuzzleResponse {
+			q := &types.KuzzleRequest{}
+			json.Unmarshal(query, q)
+
+			assert.Equal(t, "index", q.Controller)
+			assert.Equal(t, "getAutoRefresh", q.Action)
+			assert.Equal(t, "index", q.Index)
+
+			return &types.KuzzleResponse{Result: []byte(`true`)}
+		},
+	}
+
+	k, _ := kuzzle.NewKuzzle(c, nil)
+	i := index.NewIndex(k)
+	_, err := i.GetAutoRefresh("index")
+
+	assert.Nil(t, err)
```

Comment:
should also test the returned result to make sure the function correctly parses Kuzzle's JSON response